hedera-node/hedera-mono-service/src/testFixtures/java/com/hedera/test/utils/TestFixturesKeyLookup.java

```diff
@@ -53,99 +38,160 @@ public TestFixturesKeyLookup(@NonNull final ReadableStates states) {
-            return withFailureReason(INVALID_ACCOUNT_ID);
+    public Account getAccountById(@NonNull AccountID accountID) {
+        final var alias = accountID.alias();
+        if (alias != null && alias.length() > 0) {
+            final var num = aliases.get(alias.asUtf8String());
+            if (num == null) {
+                return null;
+            } else {
+                final var account = accounts.get(new EntityNumVirtualKey(num));
+                return account == null ? null : new StubbedAccount(num, alias, account);
+            }
+        } else if (!accountID.hasAccountNum()) {
+            return null;
         } else {
-            return Optional.of(account.getAccountKey())
-                    .map(key -> validateKey(key, false))
-                    .filter(reason -> reason.failed() || account.isReceiverSigRequired())
-                    .orElse(PRESENT_BUT_NOT_REQUIRED);
+            final long num = accountID.accountNumOrThrow();
+            final var account = accounts.get(new EntityNumVirtualKey(num));
+            return account == null ? null : new StubbedAccount(num, Bytes.EMPTY, account);
         }
     }
 
-    @Override
-    public KeyOrLookupFailureReason getKey(ContractID idOrAlias) {
-        final var account = accounts.get(accountNumOf(asAccount(idOrAlias)));
-        if (account == null) {
-            return withFailureReason(INVALID_CONTRACT_ID);
-        } else if (account.isDeleted() || !account.isSmartContract()) {
-            return withFailureReason(INVALID_CONTRACT_ID);
-        }
-        return validateKey(account.getAccountKey(), true);
-    }
+    private static final class StubbedAccount implements Account {
+        private final long num;
+        private final Bytes alias;
+        private final HederaAccount account;
 
-    private AccountID asAccount(final ContractID idOrAlias) {
-        return new AccountID.Builder()
-                .realmNum(idOrAlias.realmNum())
-                .shardNum(idOrAlias.shardNum())
-                .accountNum(idOrAlias.contractNumOrElse(0L))
-                .build();
-    }
+        private StubbedAccount(long num, Bytes alias, HederaAccount account) {
+            this.num = num;
+            this.alias = alias;
+            this.account = account;
+        }
 
-    @Override
-    public KeyOrLookupFailureReason getKeyIfReceiverSigRequired(ContractID idOrAlias) {
-        final var account = accounts.get(accountNumOf(asAccount(idOrAlias)));
-        if (account == null || account.isDeleted() || !account.isSmartContract()) {
-            return withFailureReason(INVALID_CONTRACT_ID);
-        } else {
-            final var key = account.getAccountKey();
-            final var keyResult = validateKey(key, true);
-            if (account.isReceiverSigRequired()) {
-                return keyResult;
-            } else {
-                return PRESENT_BUT_NOT_REQUIRED;
-            }
+        @Override
+        public long accountNumber() {
```

Comment:
Method names are inconsistent, sometimes using getFoo() semantics and sometimes foo()